notebooks/pcts_census_utils.py

```diff
@@ -143,27 +144,118 @@ def income_percentiles(row, percentiles, prefix="total"):
 ## PCTS functions
 #---------------------------------------------------------------------------------------#
-# Subset PCTS given a start date and a list of prefixes or suffixes
-def subset_pcts(start_date, prefix_and_suffix_list):
+FULL_PREFIX_LIST = ['AA','ADM',
+    'APCC', 'APCE', 'APCH', 'APCNV', 'APCS', 'APCSV', 'APCW',
+    'CHC', 'CPC', 'DIR', 'ENV', 'HPO', 'PAR', 'PS', 'TT', 'VTT', 'ZA'
+    ]
+# PM is a prefix that appears in the dummy variables as a prefix, but is not in the list of 
+# prefixes or suffixes
+
+FULL_SUFFIX_LIST = ['1A', '2A', 'AC', 'ACI', 'ADD1', 'ADU', 'AIC', 
+        'BL', 'BSA', 'CA', 'CASP', 'CATEX', 'CC', 'CC1', 'CC3', 'CCMP', 'CDO', 
+        'CDP', 'CE', 'CEX', 'CLQ', 'CM', 'CN', 'COA', 'COC', 'CPIO', 'CPIOA', 
+        'CPIOC', 'CPIOE', 'CPU', 'CR', 'CRA', 'CU', 'CUB', 'CUC', 'CUE', 'CUW', 'CUX', 'CUZ', 
+        'CWC', 'CWNC', 'DA', 'DB', 'DD', 'DEM', 'DI', 'DPS', 'DRB', 'EAF', 'EIR', 'ELD', 
+        'EXT', 'EXT2', 'EXT3', 'EXT4', 'F', 'GB', 'GPA', 'GPAJ', 'HCA', 'HCM', 'HD', 'HPOZ', 
+        'ICO', 'INT', 'M1', 'M2', 'M3', 'M6', 'M7', 'M8', 'M9', 'M10', 'M11', 'MA', 'MAEX',
+        'MCUP', 'MEL', 'MND', 'MPA', 'MPC', 'MPR', 'MSC', 'MSP', 'NC', 'ND', 'NR',
+        'O', 'OVR', 'P', 'PA', 'PA1', 'PA2', 'PA3', 'PA4', 'PA5', 'PA6', 'PA7', 'PA9', 'PA10',
+        'PA15', 'PA16', 'PA17', 'PAB', 'PAD', 'PMEX', 'PMLA', 'PMW', 'POD', 
+        'PP', 'PPR', 'PPSP', 'PSH', 'PUB', 'QC', 'RAO', 'RDP', 'RDPA',
+        'REC1', 'REC2', 'REC3', 'REC4', 'REC5', 'REV', 'RFA', 'RV',
+        'SCEA', 'SCPE', 'SE', 'SIP', 'SL', 'SLD', 'SM', 'SN', 'SP', 'SPE', 'SPP', 'SPPA', 'SPPM', 
+        'SPR', 'SUD', 'SUP1', 'TC', 'TDR', 'TOC', 'UAIZ', 'UDU', 'VCU', 'VSO', 'VZC', 'VZCJ',
+        'WDI', 'WTM', 'YV', 'ZAA', 'ZAD', 'ZAI', 'ZBA', 'ZC', 'ZCJ', 'ZV', 
+        ]
+
+# Subset PCTS and only get parent cases
+def subset_pcts_drop_child(start_date, prefix_list=None, suffix_list=None):
     """
     start_date: str with form YYYY-MM, such as "2017-10"
-    prefixes_or_suffixes: list
+    prefix_list: list of prefixes to keep. If None, then default FULL_PREFIX_LIST is used. 
+    suffix_list: list of suffixes. If None, then default FULL_SUFFIX_LIST is used.
+    
+    The sub-functions `subset_pcts` and `drop_child_cases` can be used separately.
+    If PCTS cases (parent + child) were to be subsetted by a list of prefixes / suffixes,
+    one could use just the `subset_pcts` function to return all cases that have those prefixes / suffixes.
     """
+
+    pcts = subset_pcts(start_date)
+    only_parents = drop_child_cases(pcts, prefix_list, suffix_list)
     
+    only_parents = (only_parents.sort_values(["CASE_ID", "AIN"])
+                .reset_index(drop=True)
+                )
+    return only_parents 
+
+
+# Subset PCTS given a start date and a list of prefixes or suffixes
+def subset_pcts(start_date, prefix_list=None, suffix_list=None):

```

Comment:
Might also be worth adding `end_date`, and making both dates optional. So if you call `subset_pcts` with no args, you just get back all the parent cases. Though, in that case, maybe a more appropriate name for the function would be something like `get_pcts`

Reply:
Adding an `end_date`. Since the PCTS actually goes back prior to 2010, something like 1980's (and data is wonky from then), our `start_date` we use is set to 1/1/2010.